tiledb/sm/array/array.cc

```diff
@@ -505,25 +547,35 @@ Status Array::reopen(uint64_t timestamp) {
 
   RETURN_NOT_OK(storage_manager_->array_reopen(
       array_uri_,
-      timestamp_,
       *encryption_key_,
       &array_schema_,
-      &fragment_metadata_));
+      &fragment_metadata_,
+      timestamp_start_,
+      timestamp_end_));
 
   return Status::Ok();
 }
 
 uint64_t Array::timestamp() const {
```

Comment:
Rename from `timestamp()` to `timestamp_end()` so that we use consistent terminology within the code below the C API.